Problems launching 2.27.0 container

### What is the bug or the crash?

I am upgrading from the 2.23 image to 2.27.0 and the container is hanging seemingly forever on the line:

```
INFO: Initializing Spring DispatcherServlet 'dispatcher'
```

Rolling back to 2.23.0 works again. I tried moving the existing data directory mount to rule out and kind of compatibility issue but it still persists.

Full log:

```
lc2-geoserver-container  |  _  __          _                  ____             _             
lc2-geoserver-container  | | |/ /__ _ _ __| |_ ___ ______ _  |  _ \  ___   ___| | _____ _ __
lc2-geoserver-container  | | ' // _` | '__| __/ _ \_  / _` | | | | |/ _ \ / __| |/ / _ \ '__|
lc2-geoserver-container  | | . \ (_| | |  | || (_) / / (_| | | |_| | (_) | (__|   <  __/ |
lc2-geoserver-container  | |_|\_\__,_|_|   \__\___/___\__,_| |____/ \___/ \___|_|\_\___|_|
lc2-geoserver-container  |
lc2-geoserver-container  |   ____           ____
lc2-geoserver-container  |  / ___| ___  ___/ ___|  ___ _ ____   _____ _ __
lc2-geoserver-container  | | |  _ / _ \/ _ \___ \ / _ \ '__\ \ / / _ \ '__|
lc2-geoserver-container  | | |_| |  __/ (_) |__) |  __/ |   \ V /  __/ |
lc2-geoserver-container  |  \____|\___|\___/____/ \___|_|    \_/ \___|_|
lc2-geoserver-container  |
lc2-geoserver-container  |  [Entrypoint] Enabling extension :  control-flow-plugin 
lc2-geoserver-container  |  [Entrypoint] Enabling extension :  csw-iso-plugin 
lc2-geoserver-container  |  [Entrypoint] Enabling extension :  csw-plugin 
lc2-geoserver-container  |  [Entrypoint] Enabling extension :  gdal-plugin 
lc2-geoserver-container  |  [Entrypoint] Enabling extension :  inspire-plugin 
lc2-geoserver-container  |  [Entrypoint] Enabling extension :  libjpeg-turbo-plugin 
lc2-geoserver-container  |  [Entrypoint] Enabling extension :  monitor-plugin 
lc2-geoserver-container  |  [Entrypoint] Enabling extension :  pyramid-plugin 
lc2-geoserver-container  |  [Entrypoint] Enabling extension :  vectortiles-plugin 
lc2-geoserver-container  |  [Entrypoint] Enabling extension :  wps-plugin 
lc2-geoserver-container  |  --------------------------------------------------------------------------------
lc2-geoserver-container  |  [Entrypoint] One or more variables needed for S3 community extensions are empty, skipping configuration of:  s3.properties 
lc2-geoserver-container  | .......+..+.............+........+.............+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*..+...+...+......+.....+...+.............+.....+...+......+......+....+...+...+...+..+.........+.+..+..........+..........
..+...+...........+.+.....+.+........+.......+..+.+.....+....+...+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*..+...........+.+...+..+....+.....+.............+..+.........+.........+.+......+.........+......+.........+..+..................+.
+........+.............+..+......+.+.....+................+.........+........+.+............+.....+...+.......+...+...........+...+............+...+...+.........+.............+..+....+.........+......+..+.......+........+...+....+..+................+...+........
.+.......................+...+....+.....+.+.........+.....+......+.........+...+...+.+.....+....+...........+......+...+.........+...+...+......+......+............+.......+...+.....+...+............+...+.........+.+.........+......+........+....+.....+......+.+
.....+....+..+.............+..+....+.....+...+.......+.........+..+.............+.....+.+.....+...+...+..................+...................+..+..........+.........+..+...+....+...+..............+.+........+.+..................+..+..........+...+........+......
.............+.....+......+....+...+...+..............+.+..+.......+.....+.+...+..+.........+......+.......+...+...+.....+.......+...........................+.....+.+.........+.....+......+...................+........+..........+........+......+...+..........+..+.........+..........+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
lc2-geoserver-container  | .....+..+......+.+.....+..........+.....+.........+....+...+..+...+...+...............+....+..+.+..+...+....+........+.+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*.................+.+.....+............+...+....+...+........+...................+..+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++*............+..+.............+......+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
lc2-geoserver-container  | -----
lc2-geoserver-container  | Importing keystore /etc/certs/letsencrypt.p12 to /etc/certs/letsencrypt.jks...
lc2-geoserver-container  | Warning:  Different store and key passwords not supported for PKCS12 KeyStores. Ignoring user-specified -destkeypass value.
lc2-geoserver-container  | /scripts/entrypoint.sh:57 0: CLUSTER_CONFIG_DIR=/opt/geoserver/data_dir/cluster/master/instance_a3dc45842308
lc2-geoserver-container  | /scripts/entrypoint.sh:58 0: MONITOR_AUDIT_PATH=/opt/geoserver/data_dir/monitoring/monitor_a3dc45842308
lc2-geoserver-container  |   [Entrypoint]   /opt/geoserver/data_dir/gwc   is nested in   /opt/geoserver/data_dir 
lc2-geoserver-container  |  [Entrypoint] Changing folder permission for:  /opt/geoserver/data_dir 
lc2-geoserver-container  | NOTE: Picked up JDK_JAVA_OPTIONS:  --add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.io=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.util.concurrent=ALL-UNNAMED --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Server version name:   Apache Tomcat
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Server built:          Feb 4 2025 20:08:08 UTC
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Server version number: 9.0.99.0
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: OS Name:               Linux
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: OS Version:            5.15.167.4-microsoft-standard-WSL2
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Architecture:          amd64
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Java Home:             /opt/java/openjdk
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: JVM Version:           17.0.14+7
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: JVM Vendor:            Eclipse Adoptium
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: CATALINA_BASE:         /usr/local/tomcat
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: CATALINA_HOME:         /usr/local/tomcat
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: --add-opens=java.base/java.lang=ALL-UNNAMED
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: --add-opens=java.base/java.io=ALL-UNNAMED
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: --add-opens=java.base/java.util=ALL-UNNAMED
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: --add-opens=java.base/java.util.concurrent=ALL-UNNAMED
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: --add-opens=java.rmi/sun.rmi.transport=ALL-UNNAMED
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Djava.util.logging.config.file=/usr/local/tomcat/conf/logging.properties
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Djava.util.logging.manager=org.apache.juli.ClassLoaderLogManager
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Djava.awt.headless=true
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Xms2G
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Xmx4G
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -XX:PerfDataSamplingInterval=500
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Dorg.geotools.referencing.forceXY=true
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -XX:SoftRefLRUPolicyMSPerMB=36000
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -XX:+UseG1GC
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -XX:MaxGCPauseMillis=200
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -XX:ParallelGCThreads=20
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -XX:ConcGCThreads=5
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -XX:InitiatingHeapOccupancyPercent=45
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Djts.overlay=ng
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Dfile.encoding=UTF8
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Duser.timezone=GMT
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Duser.language=en_US:en
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Duser.region=US
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Duser.country=US
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DENABLE_JSONP=true
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DMAX_FILTER_RULES=20
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DOPTIMIZE_LINE_WIDTH=false
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DALLOW_ENV_PARAMETRIZATION=false
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Djavax.servlet.request.encoding=UTF-8
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Djavax.servlet.response.encoding=UTF-8
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DCLUSTER_CONFIG_DIR=/opt/geoserver/data_dir/cluster/master/instance_a3dc45842308
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DGEOSERVER_DATA_DIR=/opt/geoserver/data_dir
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DGEOSERVER_FILEBROWSER_HIDEFS=false
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DGEOSERVER_AUDIT_PATH=/opt/geoserver/data_dir/monitoring/monitor_a3dc45842308
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Dorg.geotools.shapefile.datetime=true
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Dorg.geotools.localDateTimeHandling=true
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Dsun.java2d.renderer.useThreadLocal=false
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Dsun.java2d.renderer.pixelsize=8192
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -XX:NewSize=300m
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Dlog4j.configuration=/usr/local/tomcat/log4j.properties
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: --patch-module=java.desktop=/usr/local/tomcat/lib/marlin.jar
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Dsun.java2d.renderer=org.marlin.pisces.PiscesRenderingEngine
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Dsun.java2d.renderer.log=true
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Dgeoserver.login.autocomplete=on
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DUPDATE_BUILT_IN_LOGGING_PROFILES=false
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DRELINQUISH_LOG4J_CONTROL=false
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DGEOSERVER_CONSOLE_DISABLED=false
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DGWC_DISKQUOTA_DISABLED=false
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DGEOSERVER_CSRF_WHITELIST=
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Dgeoserver.xframe.shouldSetPolicy=true
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DGEOSERVER_REQUIRE_FILE=
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DENTITY_RESOLUTION_ALLOWLIST=www.w3.org|schemas.opengis.net|www.opengis.net|inspire.ec.europa.eu/schemas
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DGEOSERVER_DISABLE_STATIC_WEB_FILES=true
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -DGEOSERVER_DATA_DIR_LOADER_ENABLED=true
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Djdk.tls.ephemeralDHKeySize=2048
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Djava.protocol.handler.pkgs=org.apache.catalina.webresources
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Dsun.io.useCanonCaches=false
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Dorg.apache.catalina.security.SecurityListener.UMASK=0027
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Dignore.endorsed.dirs=
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Dcatalina.base=/usr/local/tomcat
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Dcatalina.home=/usr/local/tomcat
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.VersionLoggerListener log
lc2-geoserver-container  | INFO: Command line argument: -Djava.io.tmpdir=/usr/local/tomcat/temp
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.core.AprLifecycleListener lifecycleEvent
lc2-geoserver-container  | INFO: Loaded Apache Tomcat Native library [1.3.1] using APR version [1.7.0].
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.core.AprLifecycleListener lifecycleEvent
lc2-geoserver-container  | INFO: APR capabilities: IPv6 [true], sendfile [true], accept filters [false], random [true], UDS [true].
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.core.AprLifecycleListener lifecycleEvent
lc2-geoserver-container  | INFO: APR/OpenSSL configuration: useAprConnector [false], useOpenSSL [true]
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.core.AprLifecycleListener initializeSSL
lc2-geoserver-container  | INFO: OpenSSL successfully initialized [OpenSSL 3.0.2 15 Mar 2022]
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.coyote.AbstractProtocol init
lc2-geoserver-container  | INFO: Initializing ProtocolHandler ["http-nio-8080"]
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.coyote.AbstractProtocol init
lc2-geoserver-container  | INFO: Initializing ProtocolHandler ["https-openssl-nio-8443"]
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.util.LifecycleBase handleSubClassException
lc2-geoserver-container  | SEVERE: Failed to initialize component [Connector["https-openssl-nio-8443"]]
lc2-geoserver-container  | org.apache.catalina.LifecycleException: Protocol handler initialization failed
lc2-geoserver-container  |      at org.apache.catalina.connector.Connector.initInternal(Connector.java:1048)
lc2-geoserver-container  |      at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:122)
lc2-geoserver-container  |      at org.apache.catalina.core.StandardService.initInternal(StandardService.java:525)
lc2-geoserver-container  |      at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:122)
lc2-geoserver-container  |      at org.apache.catalina.core.StandardServer.initInternal(StandardServer.java:986)
lc2-geoserver-container  |      at org.apache.catalina.util.LifecycleBase.init(LifecycleBase.java:122)
lc2-geoserver-container  |      at org.apache.catalina.startup.Catalina.load(Catalina.java:690)
lc2-geoserver-container  |      at org.apache.catalina.startup.Catalina.load(Catalina.java:713)
lc2-geoserver-container  |      at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
lc2-geoserver-container  |      at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
lc2-geoserver-container  |      at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
lc2-geoserver-container  |      at java.base/java.lang.reflect.Method.invoke(Method.java:569)
lc2-geoserver-container  |      at org.apache.catalina.startup.Bootstrap.load(Bootstrap.java:302)
lc2-geoserver-container  |      at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:472)
lc2-geoserver-container  | Caused by: java.lang.IllegalArgumentException: keystore password was incorrect
lc2-geoserver-container  |      at org.apache.tomcat.util.net.AbstractJsseEndpoint.createSSLContext(AbstractJsseEndpoint.java:115)
lc2-geoserver-container  |      at org.apache.tomcat.util.net.AbstractJsseEndpoint.initialiseSsl(AbstractJsseEndpoint.java:71)
lc2-geoserver-container  |      at org.apache.tomcat.util.net.NioEndpoint.bind(NioEndpoint.java:228)
lc2-geoserver-container  |      at org.apache.tomcat.util.net.AbstractEndpoint.bindWithCleanup(AbstractEndpoint.java:1373)
lc2-geoserver-container  |      at org.apache.tomcat.util.net.AbstractEndpoint.init(AbstractEndpoint.java:1386)
lc2-geoserver-container  |      at org.apache.coyote.AbstractProtocol.init(AbstractProtocol.java:663)
lc2-geoserver-container  |      at org.apache.coyote.http11.AbstractHttp11Protocol.init(AbstractHttp11Protocol.java:80)
lc2-geoserver-container  |      at org.apache.catalina.connector.Connector.initInternal(Connector.java:1046)
lc2-geoserver-container  |      ... 13 more
lc2-geoserver-container  | Caused by: java.io.IOException: keystore password was incorrect
lc2-geoserver-container  |      at java.base/sun.security.pkcs12.PKCS12KeyStore.engineLoad(PKCS12KeyStore.java:2159)
lc2-geoserver-container  |      at java.base/sun.security.util.KeyStoreDelegator.engineLoad(KeyStoreDelegator.java:242)
lc2-geoserver-container  |      at java.base/java.security.KeyStore.load(KeyStore.java:1473)
lc2-geoserver-container  |      at org.apache.tomcat.util.security.KeyStoreUtil.load(KeyStoreUtil.java:69)
lc2-geoserver-container  |      at org.apache.tomcat.util.net.SSLUtilBase.getStore(SSLUtilBase.java:240)
lc2-geoserver-container  |      at org.apache.tomcat.util.net.SSLHostConfigCertificate.getCertificateKeystore(SSLHostConfigCertificate.java:254)
lc2-geoserver-container  |      at org.apache.tomcat.util.net.SSLUtilBase.getKeyManagers(SSLUtilBase.java:308)
lc2-geoserver-container  |      at org.apache.tomcat.util.net.openssl.OpenSSLUtil.getKeyManagers(OpenSSLUtil.java:108)
lc2-geoserver-container  |      at org.apache.tomcat.util.net.SSLUtilBase.createSSLContext(SSLUtilBase.java:268)
lc2-geoserver-container  |      at org.apache.tomcat.util.net.AbstractJsseEndpoint.createSSLContext(AbstractJsseEndpoint.java:113)
lc2-geoserver-container  |      ... 20 more
lc2-geoserver-container  | Caused by: java.security.UnrecoverableKeyException: failed to decrypt safe contents entry: javax.crypto.BadPaddingException: Given final block not properly padded. Such issues can arise if a bad key is used during decryption.
lc2-geoserver-container  |      ... 30 more
lc2-geoserver-container  |
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.Catalina load
lc2-geoserver-container  | INFO: Server initialization in [595] milliseconds
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.core.StandardService startInternal
lc2-geoserver-container  | INFO: Starting service [Catalina]
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.core.StandardEngine startInternal
lc2-geoserver-container  | INFO: Starting Servlet engine: [Apache Tomcat]
lc2-geoserver-container  | Apr 28, 2025 11:30:56 AM org.apache.catalina.startup.HostConfig deployDirectory
lc2-geoserver-container  | INFO: Deploying web application directory [/usr/local/tomcat/webapps/geoserver]
lc2-geoserver-container  | Apr 28, 2025 11:31:00 AM org.apache.jasper.servlet.TldScanner scanJars
lc2-geoserver-container  | INFO: At least one JAR was scanned for TLDs yet contained no TLDs. Enable debug logging for this logger for a complete list of JARs that were scanned but no TLDs were found in them. Skipping unneeded JARs during scanning can improve startup time and JSP compilation time.
lc2-geoserver-container  | Apr 28, 2025 11:31:00 AM org.apache.catalina.core.ApplicationContext log
lc2-geoserver-container  | INFO: No Spring WebApplicationInitializer types detected on classpath
lc2-geoserver-container  | CONFIG 2025-04-28 11:31:01,405 [main][Log4J2Logger.java:211] : Logging policy: Log4J2
lc2-geoserver-container  | CONFIG 2025-04-28 11:31:01,515 [main][Log4J2Logger.java:185] : Factory implementations for category GridCoverageFactory:
lc2-geoserver-container  |   org.geotools.coverage.grid.GridCoverageFactory
lc2-geoserver-container  | Native library load failed.
lc2-geoserver-container  | java.lang.UnsatisfiedLinkError: /usr/lib/x86_64-linux-gnu/jni/libgdalalljni.so: libgdal.so.34: cannot open shared object file: No such file or directory
lc2-geoserver-container  | Apr 28, 2025 11:31:01 AM it.geosolutions.imageio.gdalframework.GDALUtilities loadGDAL
lc2-geoserver-container  | WARNING: Failed to load the GDAL native libs. This is not a problem unless you need to use the GDAL plugins: they won't be enabled.
lc2-geoserver-container  | java.lang.UnsatisfiedLinkError: 'void org.gdal.gdal.gdalJNI.AllRegister()'
lc2-geoserver-container  | Apr 28, 2025 11:31:01 AM it.geosolutions.imageio.plugins.turbojpeg.TurboJpegUtilities load
lc2-geoserver-container  | INFO: TurboJPEG library loaded (turbojpeg)
lc2-geoserver-container  | CONFIG 2025-04-28 11:31:01,637 [main][Log4J2Logger.java:185] : Factory implementations for category FilterFactory:
lc2-geoserver-container  |   org.geotools.filter.FilterFactoryImpl
lc2-geoserver-container  | CONFIG 2025-04-28 11:31:01,650 [main][Log4J2Logger.java:185] : Factory implementations for category StyleFactory:
lc2-geoserver-container  |   org.geotools.styling.StyleFactoryImpl
lc2-geoserver-container  | CONFIG 2025-04-28 11:31:01,654 [main][Log4J2Logger.java:185] : Factory implementations for category FeatureFactory:
lc2-geoserver-container  |   org.geotools.feature.LenientFeatureFactoryImpl
lc2-geoserver-container  |   org.geotools.feature.ValidatingFeatureFactoryImpl
lc2-geoserver-container  | DEBUG 2025-04-28 11:31:01,874 [main][Log4J2Logger.java:216] : Extension lookup 'XStreamPersisterInitializer', prior to bean geoserverExtensions initialisation.
lc2-geoserver-container  | DEBUG 2025-04-28 11:31:01,875 [main][Log4J2Logger.java:216] : Extension lookup 'ExtensionFilter', prior to bean geoserverExtensions initialisation.
lc2-geoserver-container  | DEBUG 2025-04-28 11:31:01,876 [main][Log4J2Logger.java:216] : Extension lookup 'ExtensionProvider', but provided ApplicationContext is unset.
lc2-geoserver-container  | DEBUG 2025-04-28 11:31:01,876 [main][Log4J2Logger.java:216] : Extension lookup 'ExtensionFilter', but provided ApplicationContext is unset.
lc2-geoserver-container  | DEBUG 2025-04-28 11:31:01,891 [main][Log4J2Logger.java:216] : CONFIGURING GEOSERVER LOGGING -------------------------
lc2-geoserver-container  | 28 Apr 11:31:01 CONFIG [geoserver.logging] - Log4j 2 configuration set to DEFAULT_LOGGING.xml
lc2-geoserver-container  | Apr 28, 2025 11:31:01 AM org.apache.catalina.core.ApplicationContext log
lc2-geoserver-container  | INFO: Initializing Spring root WebApplicationContext
lc2-geoserver-container  | 28 Apr 11:31:03 CONFIG [org.geoserver] - GeoServer configuration lock is enabled
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [geoserver.config] - Loading catalog /opt/geoserver/data_dir
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [config.datadir] - Catalog and configuration loader uses 16 threads out of 28 available cores.
lc2-geoserver-container  | 28 Apr 11:31:04 INFO   [geoserver.config] - Read Catalog in 81.58 ms: workspaces: 0, namespaces: 0, styles: 5, stores: 0, resources: 0, layers: 0, layer groups: 0.
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [config.datadir] - Loading GeoServer config...
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [config.datadir] - Loaded service 'WFSInfoImpl--70ddb017:1967c2a3a22:-7ffb', enabled
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [config.datadir] - Loaded service 'WMSInfoImpl--70ddb017:1967c2a3a22:-7ffa', enabled
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [config.datadir] - Loaded service 'WMTSInfoImpl--70ddb017:1967c2a3a22:-7ff9', enabled
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [config.datadir] - Loaded service 'WCSInfoImpl--70ddb017:1967c2a3a22:-7ff8', enabled
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [config.datadir] - Loaded service 'WPSInfoImpl--70ddb017:1967c2a3a22:-7ff7', enabled
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [config.datadir] - Loaded service 'csw', enabled
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [config.datadir] - Catalog and configuration loader uses 16 threads out of 28 available cores.
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [config.datadir] - Loading workspace services and settings...
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [config.datadir] - GeoServer config (settings and services) loaded in 88.18 ms
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [gwc.config] - Will look for 'geowebcache.xml' in directory '/opt/geoserver/data_dir/gwc'.
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [gwc.config] - Create configuration file '/opt/geoserver/data_dir/gwc/geowebcache.xml' from template jar:file:/usr/local/tomcat/webapps/geoserver/WEB-INF/lib/gwc-core-1.27.0.jar!/geowebcache_empty.xml
lc2-geoserver-container  | 28 Apr 11:31:04 WARN   [config.XMLConfiguration] - GWC configuration validation error: cvc-elt.1.a: Cannot find the declaration of element 'gwcConfiguration'.
lc2-geoserver-container  | 28 Apr 11:31:04 WARN   [config.XMLConfiguration] - Will try to use configuration anyway. Please check the order of declared elements against the schema.
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [config.XMLConfiguration] - Initializing GridSets from /opt/geoserver/data_dir/gwc
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [config.XMLConfiguration] - Initializing layers from /opt/geoserver/data_dir/gwc
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [gwc.layer] - Initializing GWC configuration based on GeoServer's Catalog
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [gwc.layer] - GeoServer TileLayer store base directory is: gwc-layers
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [gwc.layer] - Loading tile layers from gwc-layers
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [gwc.layer] - Loaded 0 tile layers in 1.124 ms
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [storage.BlobStoreAggregator] - BlobStoreConfiguration /opt/geoserver/data_dir/gwc contained no blob store infos.
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [storage.DefaultStorageFinder] - Found system environment variableGEOWEBCACHE_CACHE_DIR set to /opt/geoserver/data_dir/gwc, using it as the default prefix.
lc2-geoserver-container  | 28 Apr 11:31:04 CONFIG [gwc.config] - Initializing GeoServer specific GWC configuration from gwc-gs.xml
lc2-geoserver-container  | 28 Apr 11:31:05 INFO   [flow.config] - Loaded control-flow rule: ows.global=100
lc2-geoserver-container  | 28 Apr 11:31:05 INFO   [flow.config] - Loaded control-flow rule: ows.wms.getmap=10
lc2-geoserver-container  | 28 Apr 11:31:05 INFO   [flow.config] - Loaded control-flow rule: ows.wfs.getfeature.application/msexcel=4
lc2-geoserver-container  | 28 Apr 11:31:05 INFO   [flow.config] - Loaded control-flow rule: user=6
lc2-geoserver-container  | 28 Apr 11:31:05 INFO   [flow.config] - Loaded control-flow rule: ows.gwc=16
lc2-geoserver-container  | 28 Apr 11:31:05 INFO   [flow.config] - Loaded control-flow rule: user.ows.wps.execute=1000/d;30s
lc2-geoserver-container  | 28 Apr 11:31:05 INFO   [flow.config] - Loaded control-flow rule: user.ows.wms.getmap=30/s
lc2-geoserver-container  | 28 Apr 11:31:05 INFO   [flow.config] - Loaded control-flow rule: ip=10
lc2-geoserver-container  | 28 Apr 11:31:05 CONFIG [geoserver.flow] - Control-flow active with 8 flow controllers
lc2-geoserver-container  | 28 Apr 11:31:05 INFO   [security.csp] - Creating csp.xml with the default configuration
lc2-geoserver-container  | 28 Apr 11:31:05 INFO   [security.csp] - Creating csp_default.xml with the default configuration
lc2-geoserver-container  | 28 Apr 11:31:05 CONFIG [gwc.config] - Will look for 'geowebcache-diskquota.xml' in directory '/opt/geoserver/data_dir/gwc'.
lc2-geoserver-container  | 28 Apr 11:31:05 CONFIG [gwc.config] - Will look for 'geowebcache-diskquota-jdbc.xml' in directory '/opt/geoserver/data_dir/gwc'.
lc2-geoserver-container  | 28 Apr 11:31:05 CONFIG [diskquota.ConfigLoader] - Quota config is: /opt/geoserver/data_dir/gwc/geowebcache-diskquota.xml
lc2-geoserver-container  | 28 Apr 11:31:05 CONFIG [diskquota.ConfigLoader] - Quota config is: /opt/geoserver/data_dir/gwc/geowebcache-diskquota.xml
lc2-geoserver-container  | 28 Apr 11:31:06 WARN   [turbojpeg.TurboJPEGMapResponse] - The turbo jpeg encoder is available for usage
lc2-geoserver-container  | 28 Apr 11:31:06 CONFIG [geoserver.security] - AuthenticationCache Initialized with 1000 Max Entries, 300 seconds idle time, 600 seconds time to live and 3 concurrency level
lc2-geoserver-container  | 28 Apr 11:31:06 CONFIG [geoserver.security] - AuthenticationCache Eviction Task created to run every 600 seconds
lc2-geoserver-container  | 28 Apr 11:31:06 WARN   [gce.imagemosaic] - Unable to set ordering between tiff readers spi
lc2-geoserver-container  | 28 Apr 11:31:07 CONFIG [geowebcache.GeoWebCacheDispatcher] - Invoked setServletPrefix(gwc)
lc2-geoserver-container  | 28 Apr 11:31:07 CONFIG [wms.WMSService] - Will NOT recombine tiles for non-tiling clients.
lc2-geoserver-container  | 28 Apr 11:31:07 CONFIG [wms.WMSService] - Will proxy requests to backend that are not getmap or getcapabilities.
lc2-geoserver-container  | 28 Apr 11:31:07 INFO   [geoserver.monitor] - Monitor extension enabled
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [geoserver.wps] - Found 5 bindable processes in GeoServer specific processes
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [geoserver.wps] - Found 89 bindable processes in Deprecated processes
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [geoserver.wps] - Blacklisting process ras:ConvolveCoverage as the input kernel of type class javax.media.jai.KernelJAI cannot be handled
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [geoserver.wps] - Blacklisting process ras:RasterZonalStatistics2 as the input zones of type class java.lang.Object cannot be handled
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [geoserver.wps] - Blacklisting process ras:RasterZonalStatistics2 as the input nodata of type class it.geosolutions.jaiext.range.Range cannot be handled
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [geoserver.wps] - Blacklisting process ras:RasterZonalStatistics2 as the input rangeData of type class java.lang.Object cannot be handled
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [geoserver.wps] - Blacklisting process ras:RasterZonalStatistics2 as the output zonal statistics of type interface java.util.List cannot be handled
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [geoserver.wps] - Found 18 bindable processes in Raster processes
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [geoserver.wps] - Blacklisting process vec:GroupCandidateSelection as the input groupingAttributes of type class java.lang.Object cannot be handled
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [geoserver.wps] - Found 31 bindable processes in Vector processes
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [geoserver.wps] - Found 48 bindable processes in Geometry processes
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [geoserver.wps] - Found 1 bindable processes in PolygonLabelProcess
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [geoserver.wps] - Found 1 bindable processes in geo
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [geoserver.wps] - Found 1 bindable processes in geo
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [diskquota.ConfigLoader] - Quota config is: /opt/geoserver/data_dir/gwc/geowebcache-diskquota.xml
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [diskquota.DiskQuotaMonitor] - Setting up disk quota periodic enforcement task
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [diskquota.DiskQuotaMonitor] - 0 layers configured with their own quotas.
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [diskquota.DiskQuotaMonitor] - 0 layers attached to global quota 20.0 GB
lc2-geoserver-container  | 28 Apr 11:31:09 CONFIG [diskquota.DiskQuotaMonitor] - Disk quota periodic enforcement task set up every 5 SECONDS
lc2-geoserver-container  | 28 Apr 11:31:10 INFO   [geoserver.security] - Strong cryptography is available
lc2-geoserver-container  | 28 Apr 11:31:10 CONFIG [geoserver.security] - Start reloading user/groups for service named default
lc2-geoserver-container  | 28 Apr 11:31:10 CONFIG [geoserver.security] - Reloading user/groups successful for service named default
lc2-geoserver-container  | 28 Apr 11:31:10 INFO   [geoserver.security] - Start encrypting configuration passwords using strongPbePasswordEncoder
lc2-geoserver-container  | 28 Apr 11:31:10 INFO   [geoserver.security] - End encrypting configuration passwords
lc2-geoserver-container  | 28 Apr 11:31:10 INFO   [config.XMLConfiguration] - Wrote configuration to /opt/geoserver/data_dir/gwc
lc2-geoserver-container  | Apr 28, 2025 11:31:10 AM org.apache.catalina.startup.HostConfig deployDirectory
lc2-geoserver-container  | INFO: Deployment of web application directory [/usr/local/tomcat/webapps/geoserver] has finished in [13,934] ms
lc2-geoserver-container  | Apr 28, 2025 11:31:10 AM org.apache.coyote.AbstractProtocol start
lc2-geoserver-container  | INFO: Starting ProtocolHandler ["http-nio-8080"]
lc2-geoserver-container  | Apr 28, 2025 11:31:10 AM org.apache.catalina.startup.Catalina start
lc2-geoserver-container  | INFO: Server startup in [13990] milliseconds
lc2-geoserver-container  | Apr 28, 2025 11:31:10 AM org.apache.catalina.core.ApplicationContext log
lc2-geoserver-container  | INFO: Initializing Spring DispatcherServlet 'dispatcher'
```

### Steps to reproduce the issue

Try to run a 2.27.0 container

### Versions

2.27.0

### Additional context

_No response_

--- a/scenario_tests/backup_restore/test.sh
+++ b/scenario_tests/backup_restore/test.sh
@@ -19,8 +19,10 @@ services=("geoserver")
 for service in "${services[@]}"; do
 
   # Execute tests
+  echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
   test_url_availability http://localhost:8080/geoserver/rest/about/version.xml
-  echo "Execute test for $service"
+  echo -e "\e[32m ---------------------------------------- \033[0m"
+  echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
   ${VERSION} -f docker-compose.yml exec $service /bin/bash /tests/test.sh
 
 done
@@ -30,8 +32,10 @@ services=("restore")
 for service in "${services[@]}"; do
 
   # Execute tests
+  echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
   test_url_availability http://localhost:8080/geoserver/rest/about/version.xml
-  echo "Execute test for $service"
+  echo -e "\e[32m ---------------------------------------- \033[0m"
+  echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
   ${VERSION} -f docker-compose.yml exec $service /bin/bash /tests/test.sh
 
 done

--- a/scenario_tests/clustering/test.sh
+++ b/scenario_tests/clustering/test.sh
@@ -26,6 +26,7 @@ services=("master")
 for service in "${services[@]}"; do
 
   # Execute tests
+  echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
   test_url_availability http://localhost:8081/geoserver/rest/about/version.xml
   echo "Execute test for $service"
   ${VERSION} -f docker-compose.yml exec "${service}" /bin/bash /tests/test.sh
@@ -38,6 +39,7 @@ services=("node")
 for service in "${services[@]}"; do
 
   # Execute tests
+  echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
   test_url_availability http://localhost:8082/geoserver/rest/about/version.xml
   echo "Execute test for $service"
   ${VERSION} -f docker-compose.yml exec "${service}" /bin/bash /tests/test.sh
@@ -67,8 +69,10 @@ services=("master")
 for service in "${services[@]}"; do
 
   # Execute tests
+  echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
   test_url_availability http://localhost:8081/geoserver/rest/about/version.xml
-  echo "Execute test for $service"
+  echo -e "\e[32m ---------------------------------------- \033[0m"
+  echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
   ${VERSION} -f docker-compose-external.yml exec "${service}" /bin/bash /tests/test.sh
 
 done
@@ -79,8 +83,10 @@ services=("node")
 for service in "${services[@]}"; do
 
   # Execute tests
+  echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
   test_url_availability http://localhost:8082/geoserver/rest/about/version.xml
-  echo "Execute test for $service"
+  echo -e "\e[32m ---------------------------------------- \033[0m"
+  echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
   ${VERSION} -f docker-compose-external.yml exec "${service}" /bin/bash /tests/test.sh
 
 done

--- a/scenario_tests/context/test.sh
+++ b/scenario_tests/context/test.sh
@@ -26,7 +26,7 @@ services=("geoserver")
 for service in "${services[@]}"; do
 
   # Execute tests
-
+  echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
   test_url_availability http://localhost:8080/foobar/rest/about/version.xml myawesomegeoserver
   echo "Execute test for $service"
   ${VERSION} exec -T "${service}" /bin/bash /tests/test.sh
@@ -48,8 +48,10 @@ services=("geoserver")
 for service in "${services[@]}"; do
 
   # Execute tests
+  echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
   test_url_availability http://localhost:8080/foobar/geoserver/rest/about/version.xml
-  echo "Execute test for $service"
+  echo -e "\e[32m ---------------------------------------- \033[0m"
+  echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
   ${VERSION} exec -T "${service}" /bin/bash /tests/test.sh
 
 done

--- a/scenario_tests/context/test.sh
+++ b/scenario_tests/context/test.sh
@@ -19,15 +19,18 @@ if [[ -n "${PRINT_TEST_LOGS}" ]]; then
 fi
 
 
-
+# Set default values
+PASS="myawesomegeoserver"
+USER="admin"
 
 services=("geoserver")
 
 for service in "${services[@]}"; do
 
+
   # Execute tests
   echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
-  test_url_availability http://localhost:8080/foobar/rest/about/version.xml myawesomegeoserver
+  test_url_availability http://localhost:8080/foobar/rest/about/version.xml "$PASS" "$USER"
   echo "Execute test for $service"
   ${VERSION} exec -T "${service}" /bin/bash /tests/test.sh
 
@@ -49,7 +52,7 @@ for service in "${services[@]}"; do
 
   # Execute tests
   echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
-  test_url_availability http://localhost:8080/foobar/geoserver/rest/about/version.xml
+  test_url_availability http://localhost:8080/foobar/geoserver/rest/about/version.xml "$PASS" "$USER"
   echo -e "\e[32m ---------------------------------------- \033[0m"
   echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
   ${VERSION} exec -T "${service}" /bin/bash /tests/test.sh

--- a/scenario_tests/disk-quota/test.sh
+++ b/scenario_tests/disk-quota/test.sh
@@ -21,8 +21,10 @@ services=("geoserver")
 for service in "${services[@]}"; do
 
   # Execute tests
+  echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
   test_url_availability http://localhost:8080/geoserver/rest/about/version.xml
-  echo "Execute test for $service"
+  echo -e "\e[32m ---------------------------------------- \033[0m"
+  echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
   ${VERSION} -f docker-compose.yml exec $service /bin/bash /tests/test.sh
 
 done

--- a/scenario_tests/gwc/test.sh
+++ b/scenario_tests/gwc/test.sh
@@ -21,9 +21,11 @@ services=("geoserver")
 for service in "${services[@]}"; do
 
   # Execute tests
+  echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
   test_url_availability http://localhost:8080/geoserver/rest/about/version.xml
   ${VERSION} -f docker-compose-gwc.yml ps
-  echo "Execute test for $service"
+  echo -e "\e[32m ---------------------------------------- \033[0m"
+  echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
   ${VERSION} -f docker-compose-gwc.yml exec  $service /bin/bash /tests/test.sh
 
 done

--- a/scenario_tests/jdbconfig/test.sh
+++ b/scenario_tests/jdbconfig/test.sh
@@ -22,8 +22,10 @@ services=("geoserver")
 for service in "${services[@]}"; do
 
   # Execute tests
+  echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
   test_url_availability http://localhost:8080/geoserver/rest/about/version.xml
-  echo "Execute test for $service"
+  echo -e "\e[32m ---------------------------------------- \033[0m"
+  echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
   ${VERSION} -f docker-compose.yml exec $service /bin/bash /tests/test.sh
 
 done
@@ -46,8 +48,10 @@ services=("geoserver")
 for service in "${services[@]}"; do
 
   # Execute tests
+  echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
   test_url_availability http://localhost:8080/geoserver/rest/about/version.xml
-  echo "Execute test for $service"
+  echo -e "\e[32m ---------------------------------------- \033[0m"
+  echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
   ${VERSION} -f docker-compose-postgis-jndi.yml exec $service /bin/bash /tests/test.sh
 
 done

--- a/scenario_tests/libjpeg/test.sh
+++ b/scenario_tests/libjpeg/test.sh
@@ -21,9 +21,11 @@ services=("geoserver")
 for service in "${services[@]}"; do
 
   # Execute tests
+  echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
   test_url_availability http://localhost:8080/geoserver/rest/about/version.xml
   ${VERSION} -f docker-compose.yml ps
-  echo "Execute test for $service"
+  echo -e "\e[32m ---------------------------------------- \033[0m"
+  echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
   ${VERSION} -f docker-compose.yml exec  $service /bin/bash /tests/test.sh
 
 done

--- a/scenario_tests/login/docker-compose.yml
+++ b/scenario_tests/login/docker-compose.yml
@@ -1,16 +1,9 @@
 
-volumes:
-  geoserver-data-dir:
-  geoserver-data-dir-pass:
-  geoserver-data-dir-cred:
-
-
 services:
   geoserver:
     image: 'kartoza/geoserver:${TAG:-manual-build}'
     restart: 'always'
     volumes:
-      - geoserver-data-dir:/opt/geoserver/data_dir
       - ./tests:/tests
     environment:
       GEOSERVER_ADMIN_PASSWORD: myawesomegeoserver
@@ -32,7 +25,6 @@ services:
     image: 'kartoza/geoserver:${TAG:-manual-build}'
     restart: 'always'
     volumes:
-      - geoserver-data-dir-pass:/opt/geoserver/data_dir
       - ./tests:/tests
     environment:
       GEOSERVER_ADMIN_USER: admin
@@ -53,7 +45,6 @@ services:
     image: 'kartoza/geoserver:${TAG:-manual-build}'
     restart: 'always'
     volumes:
-      - geoserver-data-dir-cred:/opt/geoserver/data_dir
       - ./tests:/tests
     environment:
       GEOSERVER_ADMIN_PASSWORD: myawesomegeoserver
@@ -70,3 +61,4 @@ services:
       interval: 1m30s
       timeout: 10s
       retries: 3
+

--- a/scenario_tests/login/docker-compose.yml
+++ b/scenario_tests/login/docker-compose.yml
@@ -70,4 +70,3 @@ services:
       interval: 1m30s
       timeout: 10s
       retries: 3
-

--- a/scenario_tests/login/docker-compose.yml
+++ b/scenario_tests/login/docker-compose.yml
@@ -2,6 +2,7 @@
 volumes:
   geoserver-data-dir:
   geoserver-data-dir-pass:
+  geoserver-data-dir-cred:
 
 
 services:
@@ -52,7 +53,7 @@ services:
     image: 'kartoza/geoserver:${TAG:-manual-build}'
     restart: 'always'
     volumes:
-      - geoserver-data-dir:/opt/geoserver/data_dir
+      - geoserver-data-dir-cred:/opt/geoserver/data_dir
       - ./tests:/tests
     environment:
       GEOSERVER_ADMIN_PASSWORD: myawesomegeoserver

--- a/scenario_tests/login/docker-compose.yml
+++ b/scenario_tests/login/docker-compose.yml
@@ -60,7 +60,7 @@ services:
       GEOSERVER_ADMIN_USER: myadmin
       INITIAL_MEMORY: 2G
       MAXIMUM_MEMORY: 4G
-      CONTAINER_NAME: geoserver
+      CONTAINER_NAME: credentials
       CONSOLE_HANDLER_LEVEL: WARNING
       TEST_CLASS: test_login.TestGeoServerREST
     ports:

--- a/scenario_tests/login/docker-compose.yml
+++ b/scenario_tests/login/docker-compose.yml
@@ -13,7 +13,7 @@ services:
       - geoserver-data-dir:/opt/geoserver/data_dir
       - ./tests:/tests
     environment:
-      GEOSERVER_ADMIN_PASSWORD: myawesomegeoserver
+      GEOSERVER_ADMIN_PASSWORD: fabulousgeoserver
       GEOSERVER_ADMIN_USER: admin
       INITIAL_MEMORY: 2G
       MAXIMUM_MEMORY: 4G
@@ -56,7 +56,7 @@ services:
       - geoserver-data-dir-cred:/opt/geoserver/data_dir
       - ./tests:/tests
     environment:
-      GEOSERVER_ADMIN_PASSWORD: myawesomegeoserver
+      GEOSERVER_ADMIN_PASSWORD: fabulousgeoserver
       GEOSERVER_ADMIN_USER: myadmin
       INITIAL_MEMORY: 2G
       MAXIMUM_MEMORY: 4G

--- a/scenario_tests/login/docker-compose.yml
+++ b/scenario_tests/login/docker-compose.yml
@@ -48,3 +48,25 @@ services:
       timeout: 10s
       retries: 3
 
+  credentials:
+    image: 'kartoza/geoserver:${TAG:-manual-build}'
+    restart: 'always'
+    volumes:
+      - geoserver-data-dir:/opt/geoserver/data_dir
+      - ./tests:/tests
+    environment:
+      GEOSERVER_ADMIN_PASSWORD: myawesomegeoserver
+      GEOSERVER_ADMIN_USER: myadmin
+      INITIAL_MEMORY: 2G
+      MAXIMUM_MEMORY: 4G
+      CONTAINER_NAME: geoserver
+      CONSOLE_HANDLER_LEVEL: WARNING
+      TEST_CLASS: test_login.TestGeoServerREST
+    ports:
+      - "8083:8080"
+    healthcheck:
+      test: [ "CMD-SHELL", "curl --fail --silent --write-out 'HTTP CODE : %{http_code}\n' --output /dev/null -u $${GEOSERVER_ADMIN_USER}:$${GEOSERVER_ADMIN_PASSWORD} http://localhost:8080/geoserver/rest/about/version.xml" ]
+      interval: 1m30s
+      timeout: 10s
+      retries: 3
+

--- a/scenario_tests/login/docker-compose.yml
+++ b/scenario_tests/login/docker-compose.yml
@@ -13,7 +13,7 @@ services:
       - geoserver-data-dir:/opt/geoserver/data_dir
       - ./tests:/tests
     environment:
-      GEOSERVER_ADMIN_PASSWORD: fabulousgeoserver
+      GEOSERVER_ADMIN_PASSWORD: myawesomegeoserver
       GEOSERVER_ADMIN_USER: admin
       INITIAL_MEMORY: 2G
       MAXIMUM_MEMORY: 4G
@@ -56,7 +56,7 @@ services:
       - geoserver-data-dir-cred:/opt/geoserver/data_dir
       - ./tests:/tests
     environment:
-      GEOSERVER_ADMIN_PASSWORD: fabulousgeoserver
+      GEOSERVER_ADMIN_PASSWORD: myawesomegeoserver
       GEOSERVER_ADMIN_USER: myadmin
       INITIAL_MEMORY: 2G
       MAXIMUM_MEMORY: 4G

--- a/scenario_tests/login/test.sh
+++ b/scenario_tests/login/test.sh
@@ -43,15 +43,11 @@ done
 
 ${VERSION} down -v
 
-# Test Updating passwords
-${VERSION} up -d geoserver
-
-${VERSION} stop
 
 # Update password
-sed -i 's/myawesomegeoserver/fabulousgeoserver/g' docker-compose.yml
+sed 's/myawesomegeoserver/fabulousgeoserver/g' docker-compose.yml > docker-compose-updated.yml
 # Bring the services up again
-${VERSION} up -d geoserver
+${VERSION} -f docker-compose-updated.yml up -d geoserver
 
 services=("geoserver")
 
@@ -63,9 +59,9 @@ for service in "${services[@]}"; do
   test_url_availability http://localhost:8081/geoserver/rest/about/version.xml fabulousgeoserver
   echo -e "\e[32m ---------------------------------------- \033[0m"
   echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
-  ${VERSION} exec -T $service /bin/bash /tests/test.sh
+  ${VERSION} -f docker-compose-updated.yml exec -T $service /bin/bash /tests/test.sh
 
 done
 
-${VERSION} down -v
-sed -i 's/fabulousgeoserver/myawesomegeoserver/g' docker-compose.yml
+${VERSION} -f docker-compose-updated.yml down -v
+rm -rf docker-compose-updated.yml

--- a/scenario_tests/login/test.sh
+++ b/scenario_tests/login/test.sh
@@ -15,28 +15,30 @@ fi
 
 
 services=("geoserver" "server" "credentials")
+START_PORT=8081
 
-for service in "${services[@]}"; do
+for i in "${!services[@]}"; do
+  service="${services[$i]}"
+  PORT=$((START_PORT + i))
 
-  # Execute tests
-  if [[ $service == 'server' ]];then
-    PORT=8082
+  # Set default values
+  PASS="myawesomegeoserver"
+  USER="admin"
+
+  # Service-specific overrides
+  if [[ "$service" == "server" ]]; then
     PASS=$(docker compose exec server cat /opt/geoserver/data_dir/security/pass.txt)
-    USER=admin
-  elif [[ $service == 'geoserver' ]];then
-    PORT=8081
-    PASS="myawesomegeoserver"
-    USER=admin
-  else
-    PORT=8083
-    PASS="myawesomegeoserver"
-    USER=myadmin
+  elif [[ "$service" == "credentials" ]]; then
+    USER="myadmin"
   fi
-  sleep 30
-  test_url_availability http://localhost:$PORT/geoserver/rest/about/version.xml ${PASS} ${USER}
-  echo "Execute test for $service"
-  ${VERSION} exec -T $service /bin/bash /tests/test.sh
 
+  sleep 30
+  echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
+  test_url_availability "http://localhost:$PORT/geoserver/rest/about/version.xml" "$PASS" "$USER"
+
+  echo -e "\e[32m ---------------------------------------- \033[0m"
+  echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
+  ${VERSION} exec -T "$service" /bin/bash /tests/test.sh
 done
 
 ${VERSION} down -v
@@ -57,8 +59,10 @@ for service in "${services[@]}"; do
 
   # Execute tests
   sleep 120
+  echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
   test_url_availability http://localhost:8081/geoserver/rest/about/version.xml fabulousgeoserver
-  echo "Execute test for $service"
+  echo -e "\e[32m ---------------------------------------- \033[0m"
+  echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
   ${VERSION} exec -T $service /bin/bash /tests/test.sh
 
 done

--- a/scenario_tests/login/test.sh
+++ b/scenario_tests/login/test.sh
@@ -23,7 +23,7 @@ for service in "${services[@]}"; do
     PORT=8082
     PASS=$(docker compose exec server cat /opt/geoserver/data_dir/security/pass.txt)
     USER=admin
-  elif [[ $service == 'geoserver' ]]
+  elif [[ $service == 'geoserver' ]];then
     PORT=8081
     PASS="myawesomegeoserver"
     USER=admin

--- a/scenario_tests/login/test.sh
+++ b/scenario_tests/login/test.sh
@@ -43,25 +43,3 @@ done
 
 ${VERSION} down -v
 
-
-# Update password
-sed 's/myawesomegeoserver/fabulousgeoserver/g' docker-compose.yml > docker-compose-updated.yml
-# Bring the services up again
-${VERSION} -f docker-compose-updated.yml up -d geoserver
-
-services=("geoserver")
-
-for service in "${services[@]}"; do
-
-  # Execute tests
-  sleep 120
-  echo -e "[Unit Test] Test URL availability for: \e[1;31m $service \033[0m"
-  test_url_availability http://localhost:8081/geoserver/rest/about/version.xml fabulousgeoserver
-  echo -e "\e[32m ---------------------------------------- \033[0m"
-  echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
-  ${VERSION} -f docker-compose-updated.yml exec -T $service /bin/bash /tests/test.sh
-
-done
-
-${VERSION} -f docker-compose-updated.yml down -v
-rm -rf docker-compose-updated.yml

--- a/scenario_tests/login/test.sh
+++ b/scenario_tests/login/test.sh
@@ -14,7 +14,7 @@ if [[ -n "${PRINT_TEST_LOGS}" ]]; then
 fi
 
 
-services=("geoserver" "server")
+services=("geoserver" "server" "credentials")
 
 for service in "${services[@]}"; do
 
@@ -22,12 +22,18 @@ for service in "${services[@]}"; do
   if [[ $service == 'server' ]];then
     PORT=8082
     PASS=$(docker compose exec server cat /opt/geoserver/data_dir/security/pass.txt)
-  else
+    USER=admin
+  elif [[ $service == 'geoserver' ]]
     PORT=8081
     PASS="myawesomegeoserver"
+    USER=admin
+  else
+    PORT=8083
+    PASS="myawesomegeoserver"
+    USER=myadmin
   fi
   sleep 30
-  test_url_availability http://localhost:$PORT/geoserver/rest/about/version.xml ${PASS}
+  test_url_availability http://localhost:$PORT/geoserver/rest/about/version.xml ${PASS} ${USER}
   echo "Execute test for $service"
   ${VERSION} exec -T $service /bin/bash /tests/test.sh
 

--- a/scenario_tests/login/tests/test_login.py
+++ b/scenario_tests/login/tests/test_login.py
@@ -9,29 +9,16 @@ class TestGeoServerREST(unittest.TestCase):
         # Login to GeoServer and get the authentication cookies
         self.base_url = 'http://localhost:8080/geoserver'
         self.login_url = f'{self.base_url}/j_spring_security_check'
+        self.username = 'admin'
+        self.password = environ['GEOSERVER_ADMIN_PASSWORD']
         self.container_name = environ['CONTAINER_NAME']
-        credential_map = {
-            'geoserver': {
-                'username': 'admin',
-                'password': environ['GEOSERVER_ADMIN_PASSWORD']
-            },
-            'credentials': {
-                'username': 'myadmin',
-                'password': environ['GEOSERVER_ADMIN_PASSWORD']
-            },
-            'server': {
-                'username': 'admin',
-                'password': self._read_password_file('/opt/geoserver/data_dir/security/pass.txt')
-            }
-        }
 
-        creds = credential_map.get(self.container_name)
-        if not creds:
-            raise ValueError(f"Unknown container: {self.container_name}")
-
-        self.username = creds['username']
-        self.password = creds['password']
-
+        if self.container_name == 'credentials':
+            self.username = 'myadmin'
+        elif self.container_name == 'server':
+            with open('/opt/geoserver/data_dir/security/pass.txt', 'r') as file:
+                file_pass = file.read()
+            self.password = file_pass.replace("\n", "")
         self.session = requests.Session()
         login_data = {
             'username': self.username,
@@ -40,10 +27,6 @@ class TestGeoServerREST(unittest.TestCase):
         }
         response = self.session.post(self.login_url, data=login_data)
         self.assertEqual(response.status_code, 200)
-
-    def _read_password_file(self, path):
-        with open(path, 'r') as file:
-            return file.read().strip()
 
     def test_rest_endpoints_accessible(self):
         # Test if the REST endpoints are accessible as a logged user

--- a/scenario_tests/login/tests/test_login.py
+++ b/scenario_tests/login/tests/test_login.py
@@ -9,13 +9,16 @@ class TestGeoServerREST(unittest.TestCase):
         # Login to GeoServer and get the authentication cookies
         self.base_url = 'http://localhost:8080/geoserver'
         self.login_url = f'{self.base_url}/j_spring_security_check'
-        self.username = 'admin'
-        self.password = environ['GEOSERVER_ADMIN_PASSWORD']
         self.container_name = environ['CONTAINER_NAME']
 
-        if self.container_name == 'credentials':
+        if self.container_name == 'geoserver':
+            self.password = environ['GEOSERVER_ADMIN_PASSWORD']
+            self.username = 'admin'
+        elif self.container_name == 'credentials':
+            self.password = environ['GEOSERVER_ADMIN_PASSWORD']
             self.username = 'myadmin'
         elif self.container_name == 'server':
+            self.username = 'admin'
             with open('/opt/geoserver/data_dir/security/pass.txt', 'r') as file:
                 file_pass = file.read()
             self.password = file_pass.replace("\n", "")

--- a/scenario_tests/login/tests/test_login.py
+++ b/scenario_tests/login/tests/test_login.py
@@ -9,15 +9,29 @@ class TestGeoServerREST(unittest.TestCase):
         # Login to GeoServer and get the authentication cookies
         self.base_url = 'http://localhost:8080/geoserver'
         self.login_url = f'{self.base_url}/j_spring_security_check'
-        self.username = 'admin'
         self.container_name = environ['CONTAINER_NAME']
+        credential_map = {
+            'geoserver': {
+                'username': 'admin',
+                'password': environ['GEOSERVER_ADMIN_PASSWORD']
+            },
+            'credentials': {
+                'username': 'myadmin',
+                'password': environ['GEOSERVER_ADMIN_PASSWORD']
+            },
+            'server': {
+                'username': 'admin',
+                'password': self._read_password_file('/opt/geoserver/data_dir/security/pass.txt')
+            }
+        }
 
-        if self.container_name == 'geoserver':
-            self.password = environ['GEOSERVER_ADMIN_PASSWORD']
-        else:
-            with open('/opt/geoserver/data_dir/security/pass.txt', 'r') as file:
-                file_pass = file.read()
-            self.password = file_pass.replace("\n", "")
+        creds = credential_map.get(self.container_name)
+        if not creds:
+            raise ValueError(f"Unknown container: {self.container_name}")
+
+        self.username = creds['username']
+        self.password = creds['password']
+
         self.session = requests.Session()
         login_data = {
             'username': self.username,
@@ -26,6 +40,10 @@ class TestGeoServerREST(unittest.TestCase):
         }
         response = self.session.post(self.login_url, data=login_data)
         self.assertEqual(response.status_code, 200)
+
+    def _read_password_file(self, path):
+        with open(path, 'r') as file:
+            return file.read().strip()
 
     def test_rest_endpoints_accessible(self):
         # Test if the REST endpoints are accessible as a logged user

--- a/scenario_tests/stores/test.sh
+++ b/scenario_tests/stores/test.sh
@@ -23,7 +23,8 @@ for service in "${services[@]}"; do
 
   # Execute tests
   test_url_availability http://localhost:8080/geoserver/rest/about/version.xml
-  echo "Execute test for $service"
+  echo -e "\e[32m ---------------------------------------- \033[0m"
+  echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
   ${VERSION} -f docker-compose-postgis-jndi.yml exec $service /bin/bash /tests/test.sh
 
 done
@@ -46,7 +47,8 @@ for service in "${services[@]}"; do
 
   # Execute tests
   test_url_availability http://localhost:8080/geoserver/rest/about/version.xml
-  echo "Execute test for $service"
+  echo -e "\e[32m ---------------------------------------- \033[0m"
+  echo -e "[Unit Test] Execute test for: \e[1;31m $service \033[0m"
   ${VERSION} -f docker-compose-gdal.yml exec $service /bin/bash /tests/test.sh
 
 done

--- a/scenario_tests/test-env.sh
+++ b/scenario_tests/test-env.sh
@@ -42,7 +42,10 @@ function test_url_availability() {
     result=$(wget -S --user ${USERNAME} --password ${PASS} --max-redirect=0 ${URL} 2>&1 | grep "HTTP/1.1 " | tail -n 1 | awk '{print $2}')
 
     if [[ $result -eq 200 ]]; then
-      echo "Rest endpoint ${URL} is accessible with the provided credentials"
+      echo "Rest endpoint ${URL} is accessible with the provided credentials ${USERNAME} and password ${PASS}"
+      if [[ -f version.xml ]];then
+        rm version.xml
+      fi
       break
     else
       echo "Access to ${URL}, with credentials username ${USERNAME} and password ${PASS} did not succeed, retrying in 5 seconds"

--- a/scenario_tests/test-env.sh
+++ b/scenario_tests/test-env.sh
@@ -39,7 +39,7 @@ function test_url_availability() {
 
 
 
-    result=$(wget -S --user admin --password ${PASS} --max-redirect=0 ${URL} 2>&1 | grep "HTTP/1.1 " | tail -n 1 | awk '{print $2}')
+    result=$(wget -S --user ${USERNAME} --password ${PASS} --max-redirect=0 ${URL} 2>&1 | grep "HTTP/1.1 " | tail -n 1 | awk '{print $2}')
 
     if [[ $result -eq 200 ]]; then
       echo "Rest endpoint ${URL} is accessible with the provided credentials"

--- a/scripts/update_passwords.sh
+++ b/scripts/update_passwords.sh
@@ -38,13 +38,9 @@ if [[ "${USE_DEFAULT_CREDENTIALS}" =~ [Ff][Aa][Ll][Ss][Ee] ]]; then
       # Get current GeoServer admin user
       file_env GEOSERVER_ADMIN_USER
 
-      # If GEOSERVER_ADMIN_USER is set (not empty), set GEOSERVER_ADMIN_DEFAULT_USER
-      if [ -n "${GEOSERVER_ADMIN_USER}" ]; then
-          export GEOSERVER_ADMIN_DEFAULT_USER="$GEOSERVER_ADMIN_USER"
-      else
-          GEOSERVER_ADMIN_USER='admin'
-          export GEOSERVER_ADMIN_DEFAULT_USER="$GEOSERVER_ADMIN_USER"
-      fi
+
+      export GEOSERVER_ADMIN_DEFAULT_USER='admin'
+
 
       # Get encrypted admin password
       #export GEOSERVER_ADMIN_DEFAULT_ENCRYPTED_PASSWORD="$(sed -n 's/.*password="\([^"]*\)".*/\1/p' ${USERS_XML})"

--- a/scripts/update_passwords.sh
+++ b/scripts/update_passwords.sh
@@ -46,7 +46,7 @@ if [[ "${USE_DEFAULT_CREDENTIALS}" =~ [Ff][Aa][Ll][Ss][Ee] ]]; then
       #export GEOSERVER_ADMIN_DEFAULT_ENCRYPTED_PASSWORD="$(sed -n 's/.*password="\([^"]*\)".*/\1/p' ${USERS_XML})"
 
       export PWD_HASH=$(make_hash $GEOSERVER_ADMIN_PASSWORD $CLASSPATH $HASHING_ALGORITHM)
-      ESCAPED_GEOSERVER_ADMIN_USER=$(printf '%s\n' "$GEOSERVER_ADMIN_DEFAULT_USER" | sed 's/[&/\]/\\&/g')
+      ESCAPED_GEOSERVER_ADMIN_USER=$(printf '%s\n' "$GEOSERVER_ADMIN_USER" | sed 's/[&/\]/\\&/g')
       ESCAPED_PWD_HASH=$(printf '%s\n' "$PWD_HASH" | sed 's/[&/\]/\\&/g')
       sed -i "s/name=\"[^\"]*\"/name=\"$ESCAPED_GEOSERVER_ADMIN_USER\"/; s/password=\"[^\"]*\"/password=\"$ESCAPED_PWD_HASH\"/" $USERS_XML
 

